examples/endian_swapper/tests/test_endian_swapper.py

```diff
@@ -223,8 +223,12 @@ async def run_test(dut, data_in=None, config_coroutine=None, idle_inserter=None,
 
     pkt_count = await tb.csr.read(1)
 
-    assert pkt_count.integer == tb.pkts_sent, "DUT recorded %d packets but tb counted %d" % (pkt_count.integer, tb.pkts_sent)
-    dut._log.info("DUT correctly counted %d packets" % pkt_count.integer)
+    assert convert_binary_to_unsigned(pkt_count) == tb.pkts_sent, (
```

Comment:
Is there a reason to not just use `.integer` here if the intent is to support both 1.X and 2.X?

Reply:
The problem is that `.integer` throws a deprecation warning on 2.0 which is hard to turn off. It wouldn't be a problem if cocotb-bus was a leaf package, then we could turn off the warning globally and call it a day. However, cocotb-bus can be used by third-party packages as well. This means that we should disable the warning on each call site. Since this functionality would be very similar in each call site, we would be tempted to extract it to a function and thus we would arrive to something like `convert_binary_to_unsigned` again.